text/0000-new-test-waiters.md

```diff
@@ -0,0 +1,204 @@
+- are any pending test waiters (more on waiters later!)
+- are any pending `jQuery.ajax` requests
+- are any pending route transitions.
+
+The settled check returns a Promise that is fulfilled when all of the above behaviors return `false`, indicating all async for each behavior is completed. These cover a vast number of async behaviors that are typical in our applications.
+
+An enhancement was added to [`ember-qunit`](https://github.com/emberjs/ember-qunit) that allowed for detecting a lack of settledness at the end of a test. This enhancement, [test isolation validation](https://github.com/emberjs/ember-qunit/blob/master/docs/TEST_ISOLATION_VALIDATION.md), evaluated the settled state once a test was considered done and reported to the user whether there were active async operations pending. This helped developers quickly identify and fix known asynchronous leaks in their tests, allowing for a more deterministic test suite.
+
+During the development of the test isolation validation feature, we discovered that most asynchronous operations used in the settled check provided good debug information that could be provided to the end user, with the exception of the existing test waiters. Those waiters only provided rudimentary information that could be exposed, specifically whether there were any active test waiters pending, but nothing more.
+
+To address this, a new addon was written to experiment on a new test waiter system that would provide a number of things (as noted above):
+
+1. A new API that's _explicit_ and _straightforward_
+1. A more robust way to gather debugging information for the test waiter
+1. Default test waiters with the ability to author your own, more complex test waiters
+
+This allows developers to utilize `ember-test-waiters` to annotate their asynchronous operations that are not tracked by an `await settled()` check, and for those annotations to provide useful debugging information in the event their async extended past the expected duration of the test.
+
+## Comparison of old waiters system to new
+
+In the old test waiters system, you would do the following:
+
+```js
+import { registerWaiter } from '@ember/test';
+
+registerWaiter(function() {
+  return myPendingTransactions() === 0;
+});
+```
+
+While reading the above is straightforward, when writing a test waiter using the old system it's easy to forget what the expected return value is: `true` or `false`. Additionally, it's a bit more cognitive overhead to derive what the intended result of the particular boolean return value is: does returning `true` result in the test waiter waiting or not?
+
+As mentioned before, there's no additional information provided via `registerWaiter`, and no way to capture stack traces at the call site. Unmanaged async that 'hangs' can cause your tests to stall and ultimately timeout. This is particularly problematic when trying to identify which of many test waiters has caused this timeout.
```

Comment:
How do you capture stack traces in the new test waiters design?

Reply:
@Gaurav0 - When `.beginAsync` is called on a test waiter we capture the stack trace (in debug builds) and save it off.

Reply:
I added a brief description of how we do it.